metricbeat monitoring of Elasticsearch and Kibana broke on recent 7.17.0-SNAPSHOT rpm

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 7.17.0-SNAPSHOT
- Operating System: CentOS 7
- Discuss Forum URL:
- Steps to Reproduce: 
1. Install metricbeat and enable modules for monitoring Elasticsearch, Kibana, Logstash, and Beats

Monitoring worked on the 7.17.0-766e7ca2 SNAPSHOT build from Jan 18, 2022 10:02 AM
but fails on the 7.17.0-68da5d12 SNAPSHOT build from Jan 18, 2022 5:38 PM
and also fails on the 7.17.0-1bd53ff7 BC2 build.

I don't find anything of interest in the metricbeat log.  I don't see anything about monitoring in the log with either the passing or failing tests.

All of the configuration is the same between the build that passes and the builds that fail.  I've run the passing and failing builds locally with the same results as on Jenkins.

/etc/metricbeat/metricbeat.yml:
```
metricbeat.config.modules:
  path: ${path.config}/modules.d/*.yml
  reload.period: 10s
  reload.enabled: false
metricbeat.max_start_delay: 10s
metricbeat.modules:
- module: system
  enabled: true
  period: 10s
  metricsets:
    - cpu
    - load
    - memory
    - network
    - process
    - process_summary
  processes: ['.*']
- module: system
  enabled: true
  period: 1m
  metricsets:
    - filesystem
    - fsstat
- module: kibana
  metricsets: ["status"]
  enabled: false
  period: 10s
  hosts: ["localhost:5601"]
- module: elasticsearch
  metricsets: ["node", "node_stats"]
  enabled: false
  period: 10s
  hosts: ["localhost:9200"]


output.elasticsearch:
  hosts: ["localhost:9200"]
  protocol: https
  username: "beats_internal"
  password: "changeme"
  ssl.enabled: true
  ssl.certificate_authorities: ["/etc/metricbeat/ca.crt"]
  ssl.certificate: /etc/metricbeat/elasticsearch.crt
  ssl.key: /etc/metricbeat/elasticsearch.key

#setup.dashboards.snapshot: true

setup.kibana:
  protocol: https
  host: localhost:5601
  username: "beats_internal"
  password: "changeme"
  ssl.enabled: true
  ssl.certificate_authorities: ["/etc/metricbeat/ca.crt"]
  ssl.certificate: /etc/metricbeat/elasticsearch.crt
  ssl.key: /etc/metricbeat/elasticsearch.key

xpack.monitoring:
  enabled: false
  elasticsearch:
    url: "https://localhost:9200"
    username: elastic
    password: changeit
http.enabled: true
http.port: 5068
```

```
[root@localhost vagrant]# ls -l /etc/metricbeat/modules.d/ | grep -v disabled
total 272
-rw-r--r-- 1 root root  295 Jan 19 22:31 beat-xpack.yml
-rw-r--r-- 1 root root  456 Jan 19 22:31 elasticsearch-xpack.yml
-rw-r--r-- 1 root root  461 Jan 19 22:31 kibana-xpack.yml
-rw-r--r-- 1 root root  251 Jan 19 22:31 logstash-xpack.yml
-rw-r--r-- 1 root root  956 Jan 19 14:31 system.yml
```

The `beats_internal` user in the config has;
```
curl -k https://elastic:changeit@localhost:9200/_security/user/beats_internal
{"beats_internal":{"username":"beats_internal","roles":["beats_writer","beats_reader","kibana_admin","machine_learning_admin","remote_monitoring_agent"],"full_name":"beats_internal","email":"beats_internal@test.co","metadata":{},"enabled":true}}
```

And `beats_reader` role is;
```
[root@localhost vagrant]# curl -k https://elastic:changeit@localhost:9200/_security/role/beats_reader
{"beats_reader":{"cluster":[],"indices":[{"names":["*beat-*"],"privileges":["read","view_index_metadata"],"allow_restricted_indices":false}],"applications":[],"run_as":[],"metadata":{},"transient_metadata":{"enabled":true}}}
```

And `beats_writer` role is;
```
[root@localhost vagrant]# curl -k https://elastic:changeit@localhost:9200/_security/role/beats_writer
{"beats_writer":{"cluster":["manage_index_templates","monitor","manage_ilm"],"indices":[{"names":["filebeat-*","metricbeat-*","heartbeat-*","packetbeat-*","auditbeat-*","winlogbeat-*",".kibana*"],"privileges":["write","delete","create_index","manage","manage_ilm"],"allow_restricted_indices":false}],"applications":[],"run_as":[],"metadata":{},"transient_metadata":{"enabled":true}}}
```

I can share full logs (maybe not on GH though as it is quite a lot and contains sensitive bits?). Initially Elasticsearch does not seem to be available yet (but comes online a few moments later) and then logs contain variations of the last two messages from this excerpt repeated ad infinitum: 
```
2022-01-26T13:27:11.404Z        INFO    instance/beat.go:686    Home path: [/usr/share/metricbeat] Config path: [/usr/share/metricbeat] Data path: [/usr/s
hare/metricbeat/data] Logs path: [/usr/share/metricbeat/logs] Hostfs Path: [/]
2022-01-26T13:27:11.408Z        INFO    instance/beat.go:694    Beat ID: b8f63808-ac58-41f2-8b3b-568f5db7e48a
2022-01-26T13:27:11.411Z        WARN    [add_cloud_metadata]    add_cloud_metadata/provider_aws_ec2.go:95       error when check request status for gettin
g IMDSv2 token: http request status 405. No token in the metadata request will be used.
2022-01-26T13:27:11.413Z        INFO    [seccomp]       seccomp/seccomp.go:124  Syscall filter successfully installed
2022-01-26T13:27:11.413Z        INFO    [beat]  instance/beat.go:1040   Beat info       {"system_info": {"beat": {"path": {"config": "/usr/share/metricbea
t", "data": "/usr/share/metricbeat/data", "home": "/usr/share/metricbeat", "logs": "/usr/share/metricbeat/logs"}, "type": "metricbeat", "uuid": "b8f63808-
ac58-41f2-8b3b-568f5db7e48a"}}}
2022-01-26T13:27:11.413Z        INFO    [beat]  instance/beat.go:1049   Build info      {"system_info": {"build": {"commit": "b55f1ceac49d295c6050c02e336b
56911f580abd", "libbeat": "7.17.0", "time": "2022-01-23T01:51:29.000Z", "version": "7.17.0"}}}
2022-01-26T13:27:11.413Z        INFO    [beat]  instance/beat.go:1052   Go runtime info {"system_info": {"go": {"os":"linux","arch":"amd64","max_procs":8,
"version":"go1.17.5"}}}
2022-01-26T13:27:11.414Z        INFO    [beat]  instance/beat.go:1056   Host info       {"system_info": {"host": {"architecture":"x86_64","boot_time":"202
2-01-04T15:48:06Z","containerized":true,"name":"test-es-mon-a-fftl-es-masterdata-0","ip":["127.0.0.1/8","10.73.50.171/24"],"kernel_version":"5.4.150+","ma
c":["d2:b5:d2:f3:3a:af"],"os":{"type":"linux","family":"debian","platform":"ubuntu","name":"Ubuntu","version":"20.04.3 LTS (Focal Fossa)","major":20,"mino
r":4,"patch":3,"codename":"focal"},"timezone":"UTC","timezone_offset_sec":0}}}
2022-01-26T13:27:11.414Z        INFO    [beat]  instance/beat.go:1085   Process info    {"system_info": {"process": {"capabilities": {"inheritable":["chow
n","dac_override","fowner","fsetid","kill","setgid","setuid","setpcap","net_bind_service","net_raw","sys_chroot","mknod","audit_write","setfcap"],"permitt
ed":null,"effective":null,"bounding":["chown","dac_override","fowner","fsetid","kill","setgid","setuid","setpcap","net_bind_service","net_raw","sys_chroot
","mknod","audit_write","setfcap"],"ambient":null}, "cwd": "/usr/share/metricbeat", "exe": "/usr/share/metricbeat/metricbeat", "name": "metricbeat", "pid"
: 7, "ppid": 1, "seccomp": {"mode":"filter","no_new_privs":true}, "start_time": "2022-01-26T13:27:10.570Z"}}}
2022-01-26T13:27:11.414Z        INFO    instance/beat.go:328    Setup Beat: metricbeat; Version: 7.17.0
2022-01-26T13:27:11.414Z        INFO    [index-management]      idxmgmt/std.go:184      Set output.elasticsearch.index to 'metricbeat-7.17.0' as ILM is en
abled.
2022-01-26T13:27:11.415Z        WARN    [cfgwarn]       tlscommon/config.go:100 DEPRECATED: Treating the CommonName field on X.509 certificates as a host 
name when no Subject Alternative Names are present is going to be removed. Please update your certificates if needed. Will be removed in version: 8.0.0
2022-01-26T13:27:11.415Z        INFO    [esclientleg]   eslegclient/connection.go:102   elasticsearch url: https://test-es-mon-metrics-p2rd-es-http.e2e-me
rcury.svc:9200
2022-01-26T13:27:11.415Z        INFO    [publisher]     pipeline/module.go:113  Beat name: test-es-mon-a-fftl-es-masterdata-0
2022-01-26T13:27:11.428Z        INFO    [add_cloud_metadata]    add_cloud_metadata/add_cloud_metadata.go:105    add_cloud_metadata: hosting provider type 
detected as gcp, metadata={"cloud":{"account":{"id":"..."},"availability_zone":"europe-west1-c","instance":{"id":"2933194355478483660","name
":"gke-pebrc-dev-cluster-default-pool-12721212-l8gk"},"machine":{"type":"n1-standard-8"},"project":{"id":"elastic-cloud-dev"},"provider":"gcp","service":{
"name":"GCE"}},"orchestrator":{"cluster":{"name":"pebrc-dev-cluster","url":"https://...."}}}
2022-01-26T13:27:11.447Z        INFO    [monitoring]    log/log.go:142  Starting metrics logging every 30s
2022-01-26T13:27:11.449Z        INFO    instance/beat.go:492    metricbeat start running.
2022-01-26T13:27:13.583Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.shard: error determining if connected Elas
ticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:13.788Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.index_summary: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp [::1]:9200: connect: cannot assign requested address
2022-01-26T13:27:14.584Z        INFO    [publisher_pipeline_output]     pipeline/output.go:143  Connecting to backoff(elasticsearch(https://test-es-mon-metrics-p2rd-es-http.e2e-mercury.svc:9200))
2022-01-26T13:27:14.584Z        INFO    [publisher]     pipeline/retry.go:219   retryer: send unwait signal to consumer
2022-01-26T13:27:14.584Z        INFO    [publisher]     pipeline/retry.go:223     done
2022-01-26T13:27:14.712Z        INFO    [esclientleg]   eslegclient/connection.go:282   Attempting to connect to Elasticsearch version 7.17.0-SNAPSHOT
2022-01-26T13:27:14.794Z        INFO    [esclientleg]   eslegclient/connection.go:282   Attempting to connect to Elasticsearch version 7.17.0-SNAPSHOT
2022-01-26T13:27:14.821Z        INFO    [index-management]      idxmgmt/std.go:261      Auto ILM enable success.
2022-01-26T13:27:14.885Z        INFO    [index-management.ilm]  ilm/std.go:180  ILM policy metricbeat successfully created.
2022-01-26T13:27:14.885Z        INFO    [index-management]      idxmgmt/std.go:401      Set setup.template.name to '{metricbeat-7.17.0 {now/d}-000001}' as
 ILM is enabled.
2022-01-26T13:27:14.885Z        INFO    [index-management]      idxmgmt/std.go:406      Set setup.template.pattern to 'metricbeat-7.17.0-*' as ILM is enab
led.
2022-01-26T13:27:14.885Z        INFO    [index-management]      idxmgmt/std.go:440      Set settings.index.lifecycle.rollover_alias in template to {metric
beat-7.17.0 {now/d}-000001} as ILM is enabled.
2022-01-26T13:27:14.885Z        INFO    [index-management]      idxmgmt/std.go:444      Set settings.index.lifecycle.name in template to {metricbeat {"pol
icy":{"phases":{"hot":{"actions":{"rollover":{"max_age":"30d","max_size":"50gb"}}}}}}} as ILM is enabled.
2022-01-26T13:27:14.892Z        INFO    template/load.go:197    Existing template will be overwritten, as overwrite is enabled.
2022-01-26T13:27:15.321Z        INFO    template/load.go:131    Try loading template metricbeat-7.17.0 to Elasticsearch
2022-01-26T13:27:15.544Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.ml_job: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:15.644Z        INFO    template/load.go:123    Template with name "metricbeat-7.17.0" loaded.
2022-01-26T13:27:15.644Z        INFO    [index-management]      idxmgmt/std.go:297      Loaded index template.
2022-01-26T13:27:15.884Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.index: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:16.302Z        INFO    [index-management.ilm]  ilm/std.go:140  Index Alias metricbeat-7.17.0 successfully created.
2022-01-26T13:27:16.304Z        INFO    [publisher_pipeline_output]     pipeline/output.go:151  Connection to backoff(elasticsearch(https://test-es-mon-metrics-p2rd-es-http.e2e-mercury.svc:9200)) established
2022-01-26T13:27:17.078Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.index_recovery: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:18.327Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.enrich: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:18.546Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.cluster_stats: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp [::1]:9200: connect: cannot assign requested address
2022-01-26T13:27:18.604Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.ccr: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp [::1]:9200: connect: cannot assign requested address
2022-01-26T13:27:20.691Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.node_stats: error making http request: Get "https://localhost:9200/_nodes/_local/stats": dial tcp [::1]:9200: connect: cannot assign requested address
2022-01-26T13:27:23.584Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.shard: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:23.790Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.index_summary: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:25.544Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.ml_job: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:25.886Z        ERROR   module/wrapper.go:259   Error fetching data for metricset elasticsearch.index: error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused
2022-01-26T13:27:28.443Z        WARN    [elasticsearch] elasticsearch/client.go:422     Cannot index event publisher.Event{Content:beat.Event{Timestamp:time.Date(2022, time.January, 26, 13, 27, 27, 79486756, time.Local), Meta:{"index":".monitoring-es-7-mb"}, Fields:{"agent":{"ephemeral_id":"623de1bf-2611-45a5-86de-dab0355950e7","hostname":"test-es-mon-a-fftl-es-masterdata-0","id":"b8f63808-ac58-41f2-8b3b-568f5db7e48a","name":"test-es-mon-a-fftl-es-masterdata-0","type":"metricbeat","version":"7.17.0"},"cloud":{"account":{"id":"...."},"availability_zone":"europe-west1-c","instance":{"id":"2933194355478483660","name":"gke-pebrc-dev-cluster-default-pool-12721212-l8gk"},"machine":{"type":"n1-standard-8"},"project":{"id":"elastic-cloud-dev"},"provider":"gcp","service":{"name":"GCE"}},"cluster_uuid":"xW1xX2dKQXuIOMK3scivoA","ecs":{"version":"1.12.0"},"event":{"dataset":"elasticsearch.index.recovery","duration":354978007,"module":"elasticsearch"},"host":{"architecture":"x86_64","containerized":true,"hostname":"test-es-mon-a-fftl-es-masterdata-0","ip":["10.73.50.171"],"mac":["d2:b5:d2:f3:3a:af"],"name":"test-es-mon-a-fftl-es-masterdata-0","os":{"codename":"focal","family":"debian","kernel":"5.4.150+","name":"Ubuntu","platform":"ubuntu","type":"linux","version":"20.04.3 LTS (Focal Fossa)"}},"index_recovery":{"shards":[{"id":0,"index":{"files":{"percent":"100.0%","recovered":0,"reused":16,"total":16},"size":{"percent":"100.0%","recovered_from_snapshot_in_bytes":0,"recovered_in_bytes":0,"reused_in_bytes":42290519,"total_in_bytes":42290519},"source_throttle_time_in_millis":0,"target_throttle_time_in_millis":0,"total_time_in_millis":4},"index_name":".geoip_databases","primary":true,"source":{"bootstrap_new_history_uuid":false},"stage":"DONE","start_time_in_millis":1643203646750,"stop_time_in_millis":1643203647213,"target":{"host":"10.73.50.171","id":"bH8-2OueRnWIKg2Ve0awYg","ip":"10.73.50.171","name":"test-es-mon-a-fftl-es-masterdata-0","transport_address":"10.73.50.171:9300"},"total_time_in_millis":463,"translog":{"percent":"100.0%","recovered":0,"total":0,"total_on_start":0,"total_time_in_millis":419},"type":"EXISTING_STORE","verify_index":{"check_index_time_in_millis":0,"total_time_in_millis":0}}]},"interval_ms":10000,"metricset":{"name":"index_recovery","period":10000},"orchestrator":{"cluster":{"name":"pebrc-dev-cluster","url":"https://...."}},"service":{"address":"https://localhost:9200","type":"elasticsearch"},"timestamp":"2022-01-26T13:27:27.434Z","type":"index_recovery"}, Private:interface {}(nil), TimeSeries:true}, Flags:0x0, Cache:publisher.EventCache{m:common.MapStr(nil)}} (status=404): {"type":"index_not_found_exception","reason":"no such index [.monitoring-es-7-mb-2022.01.26] and [require_alias] request flag is [true] and [.monitoring-es-7-mb-2022.01.26] is not an alias","index_uuid":"_na_","index":".monitoring-es-7-mb-2022.01.26"}, dropping event!
2022-01-26T13:27:28.443Z        WARN    [elasticsearch] elasticsearch/client.go:422     Cannot index event publisher.Event{Content:beat.Event{Timestamp:time.Date(2022, time.January, 26, 13, 27, 28, 328201889, time.Local), Meta:{"index":".monitoring-es-7-mb"}, Fields:{"agent":{"ephemeral_id":"623de1bf-2611-45a5-86de-dab0355950e7","hostname":"test-es-mon-a-fftl-es-masterdata-0","id":"b8f63808-ac58-41f2-8b3b-568f5db7e48a","name":"test-es-mon-a-fftl-es-masterdata-0","type":"metricbeat","version":"7.17.0"},"cloud":{"account":{"id":"...."},"availability_zone":"europe-west1-c","instance":{"id":"2933194355478483660","name":"gke-pebrc-dev-cluster-default-pool-12721212-l8gk"},"machine":{"type":"n1-standard-8"},"project":{"id":"elastic-cloud-dev"},"provider":"gcp","service":{"name":"GCE"}},"cluster_uuid":"xW1xX2dKQXuIOMK3scivoA","ecs":{"version":"1.12.0"},"enrich_coordinator_stats":{"executed_searches_total":0,"node_id":"bH8-2OueRnWIKg2Ve0awYg","queue_size":0,"remote_requests_current":0,"remote_requests_total":0},"event":{"dataset":"elasticsearch.enrich","duration":102546677,"module":"elasticsearch"},"host":{"architecture":"x86_64","containerized":true,"hostname":"test-es-mon-a-fftl-es-masterdata-0","ip":["10.73.50.171"],"mac":["d2:b5:d2:f3:3a:af"],"name":"test-es-mon-a-fftl-es-masterdata-0","os":{"codename":"focal","family":"debian","kernel":"5.4.150+","name":"Ubuntu","platform":"ubuntu","type":"linux","version":"20.04.3 LTS (Focal Fossa)"}},"interval_ms":10000,"metricset":{"name":"enrich","period":10000},"orchestrator":{"cluster":{"name":"pebrc-dev-cluster","url":"https://....."}},"service":{"address":"https://localhost:9200","type":"elasticsearch"},"timestamp":"2022-01-26T13:27:28.430Z","type":"enrich_coordinator_stats"}, Private:interface {}(nil), TimeSeries:true}, Flags:0x0, Cache:publisher.EventCache{m:common.MapStr(nil)}} (status=404): {"type":"index_not_found_exception","reason":"no such index [.monitoring-es-7-mb-2022.01.26] and [require_alias] request flag is [true] and [.monitoring-es-7-mb-2022.01.26] is not an alias","index_uuid":"_na_","index":".monitoring-es-7-mb-2022.01.26"}, dropping event!
``` 

I can also share the metricbeat configuration if that helps.

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -39,8 +39,6 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 
 *Affecting all Beats*
 
-- Enable `require_alias` for Bulk requests for all actions when target is a write alias. {issue}27874[27874] {pull}29879[29879]
-
 
 *Auditbeat*
 

--- a/libbeat/esleg/eslegclient/bulkapi.go
+++ b/libbeat/esleg/eslegclient/bulkapi.go
@@ -51,11 +51,10 @@ type BulkDeleteAction struct {
 }
 
 type BulkMeta struct {
-	Index        string `json:"_index" struct:"_index"`
-	DocType      string `json:"_type,omitempty" struct:"_type,omitempty"`
-	Pipeline     string `json:"pipeline,omitempty" struct:"pipeline,omitempty"`
-	ID           string `json:"_id,omitempty" struct:"_id,omitempty"`
-	RequireAlias bool   `json:"require_alias,omitempty" struct:"require_alias,omitempty"`
+	Index    string `json:"_index" struct:"_index"`
+	DocType  string `json:"_type,omitempty" struct:"_type,omitempty"`
+	Pipeline string `json:"pipeline,omitempty" struct:"pipeline,omitempty"`
+	ID       string `json:"_id,omitempty" struct:"_id,omitempty"`
 }
 
 type bulkRequest struct {

--- a/libbeat/idxmgmt/std.go
+++ b/libbeat/idxmgmt/std.go
@@ -337,16 +337,12 @@ func (s *ilmIndexSelector) Select(evt *beat.Event) (string, error) {
 	return idx, err
 }
 
-func (s ilmIndexSelector) IsAlias() bool { return true }
-
 func (s indexSelector) Select(evt *beat.Event) (string, error) {
 	if idx := getEventCustomIndex(evt, s.beatInfo); idx != "" {
 		return idx, nil
 	}
 	return s.sel.Select(evt)
 }
-
-func (s indexSelector) IsAlias() bool { return false }
 
 func getEventCustomIndex(evt *beat.Event, beatInfo beat.Info) string {
 	if len(evt.Meta) == 0 {

--- a/libbeat/outputs/elasticsearch/client.go
+++ b/libbeat/outputs/elasticsearch/client.go
@@ -317,10 +317,6 @@ func (client *Client) createEventBulkMeta(version common.Version, event *beat.Ev
 		ID:       id,
 	}
 
-	if isRequireAliasSupported(version) {
-		meta.RequireAlias = client.index.IsAlias()
-	}
-
 	if opType == events.OpTypeDelete {
 		if id != "" {
 			return eslegclient.BulkDeleteAction{Delete: meta}, nil
@@ -335,10 +331,6 @@ func (client *Client) createEventBulkMeta(version common.Version, event *beat.Ev
 		return eslegclient.BulkCreateAction{Create: meta}, nil
 	}
 	return eslegclient.BulkIndexAction{Index: meta}, nil
-}
-
-func isRequireAliasSupported(version common.Version) bool {
-	return !version.LessThan(common.MustNewVersion("7.10.0"))
 }
 
 func (client *Client) getPipeline(event *beat.Event) (string, error) {

--- a/libbeat/outputs/elasticsearch/client_integration_test.go
+++ b/libbeat/outputs/elasticsearch/client_integration_test.go
@@ -420,8 +420,7 @@ func connectTestEs(t *testing.T, cfg interface{}, stats outputs.Observer) (outpu
 	}
 
 	info := beat.Info{Beat: "libbeat"}
-	// ILM must be disabled otherwise custom index settings are ignored.
-	im, _ := idxmgmt.DefaultSupport(nil, info, disabledILMConfig())
+	im, _ := idxmgmt.DefaultSupport(nil, info, nil)
 	output, err := makeES(im, info, stats, config)
 	if err != nil {
 		t.Fatal(err)
@@ -437,10 +436,6 @@ func connectTestEs(t *testing.T, cfg interface{}, stats outputs.Observer) (outpu
 	client.Connect()
 
 	return client, client
-}
-
-func disabledILMConfig() *common.Config {
-	return common.MustNewConfigFrom(map[string]interface{}{"setup": map[string]interface{}{"ilm": map[string]interface{}{"enabled": false}}})
 }
 
 // setupRoleMapping sets up role mapping for the Kerberos user beats@ELASTIC

--- a/libbeat/outputs/elasticsearch/client_test.go
+++ b/libbeat/outputs/elasticsearch/client_test.go
@@ -409,49 +409,22 @@ func TestClientWithHeaders(t *testing.T) {
 
 func TestBulkEncodeEvents(t *testing.T) {
 	cases := map[string]struct {
-		version   string
-		docType   string
-		config    common.MapStr
-		ilmConfig *common.Config
-		isAlias   bool
-		events    []common.MapStr
+		version string
+		docType string
+		config  common.MapStr
+		events  []common.MapStr
 	}{
 		"6.x": {
-			version:   "6.8.0",
-			docType:   "doc",
-			config:    common.MapStr{},
-			ilmConfig: common.NewConfig(),
-			events:    []common.MapStr{{"message": "test"}},
+			version: "6.8.0",
+			docType: "doc",
+			config:  common.MapStr{},
+			events:  []common.MapStr{{"message": "test"}},
 		},
-		"require_alias not supported": {
-			version:   "7.9.0",
-			docType:   "",
-			config:    common.MapStr{},
-			ilmConfig: common.NewConfig(),
-			events:    []common.MapStr{{"message": "test"}},
-		},
-		"require_alias is supported": {
-			version:   "7.10.0",
-			docType:   "",
-			config:    common.MapStr{},
-			ilmConfig: common.NewConfig(),
-			isAlias:   true,
-			events:    []common.MapStr{{"message": "test"}},
-		},
-		"latest with ILM": {
-			version:   version.GetDefaultVersion(),
-			docType:   "",
-			config:    common.MapStr{},
-			ilmConfig: common.NewConfig(),
-			isAlias:   true,
-			events:    []common.MapStr{{"message": "test"}},
-		},
-		"latest without ILM": {
-			version:   version.GetDefaultVersion(),
-			docType:   "",
-			config:    common.MapStr{},
-			ilmConfig: disabledILMConfig(),
-			events:    []common.MapStr{{"message": "test"}},
+		"latest": {
+			version: version.GetDefaultVersion(),
+			docType: "",
+			config:  common.MapStr{},
+			events:  []common.MapStr{{"message": "test"}},
 		},
 	}
 
@@ -464,7 +437,7 @@ func TestBulkEncodeEvents(t *testing.T) {
 				Version:     test.version,
 			}
 
-			im, err := idxmgmt.DefaultSupport(nil, info, test.ilmConfig)
+			im, err := idxmgmt.DefaultSupport(nil, info, common.NewConfig())
 			require.NoError(t, err)
 
 			index, pipeline, err := buildSelectors(im, info, cfg)
@@ -506,17 +479,12 @@ func TestBulkEncodeEvents(t *testing.T) {
 				}
 
 				assert.NotEqual(t, "", meta.Index)
-				assert.Equal(t, test.isAlias, meta.RequireAlias)
 				assert.Equal(t, test.docType, meta.DocType)
 			}
 
 			// TODO: customer per test case validation
 		})
 	}
-}
-
-func disabledILMConfig() *common.Config {
-	return common.MustNewConfigFrom(map[string]interface{}{"setup": map[string]interface{}{"ilm": map[string]interface{}{"enabled": false}}})
 }
 
 func TestBulkEncodeEventsWithOpType(t *testing.T) {

--- a/libbeat/outputs/elasticsearch/death_letter_selector.go
+++ b/libbeat/outputs/elasticsearch/death_letter_selector.go
@@ -34,5 +34,3 @@ func (d DeadLetterSelector) Select(event *beat.Event) (string, error) {
 	}
 	return d.Selector.Select(event)
 }
-
-func (d DeadLetterSelector) IsAlias() bool { return false }

--- a/libbeat/outputs/outil/select.go
+++ b/libbeat/outputs/outil/select.go
@@ -87,8 +87,6 @@ func (s Selector) Select(evt *beat.Event) (string, error) {
 	return s.sel.sel(evt)
 }
 
-func (s Selector) IsAlias() bool { return false }
-
 // IsEmpty checks if the selector is not configured and will always return an empty string.
 func (s Selector) IsEmpty() bool {
 	return s.sel == nilSelector || s.sel == nil

--- a/libbeat/outputs/output_reg.go
+++ b/libbeat/outputs/output_reg.go
@@ -43,10 +43,8 @@ type IndexManager interface {
 }
 
 // IndexSelector is used to find the index name an event shall be indexed to.
-// It also used to check if during indexing required_alias should be set.
 type IndexSelector interface {
 	Select(event *beat.Event) (string, error)
-	IsAlias() bool
 }
 
 // Group configures and combines multiple clients into load-balanced group of clients

